content/references/protocol-reference/transactions/transaction-types/ammdeposit.md

```diff
@@ -0,0 +1,131 @@
+- **Single-asset deposits**, in which you provide only one of the AMM's two assets. You can envision a single-asset deposit as trading _some_ of the deposited asset for the other asset, then performing a double-asset deposit. The trading fee applies to the amount you would need to trade for, but not to the rest of the deposit; the amount of the fee is debited from the LP Tokens paid out. _For example, if the AMM's asset pool is split perfectly evenly between USD and EUR, and you try to deposit 100 USD, the amount of LP Tokens you receive is slightly less than if you had deposited 50 EUR + 50 USD, because you pay the trading fee to convert some of your USD to an equal amount of EUR._
+
+The following combinations of fields indicate a **double-asset deposit**:
+
+| Flag Name    | Flag Value   | Fields Specified       | Meaning |
+|--------------|--------------|------------------------|---------|
+| `tfLPToken`  | `0x00010000` | `LPTokenOut` only      | Deposit both of this AMM's assets, in amounts calculated so that you receive the specified amount of LP Tokens in return. The amounts deposited maintain the relative proportions of the two assets the AMM already holds. |
+| `tfTwoAsset` | `0x00100000` | `Amount` and `Amount2` | Deposit both of this AMM's assets, up to the specified amounts. The actual amounts deposited must maintain the same balance of assets as the AMM already holds, so the amount of either one deposited MAY be less than specified. The amount of LP Tokens you get in return is based on the total value deposited. |
+
+The following combinations of fields indicate a **single asset deposit**:
+
+| Flag Name           | Flag Value   | Fields Specified           | Meaning |
+|---------------------|--------------|----------------------------|---------|
+| `tfSingleAsset`     | `0x00080000` | `Amount` only              | Deposit exactly the specified amount of one asset, and receive an amount of LP Tokens based on the resulting share of the pool (minus fees). |
+| `tfOneAssetLPToken` | `0x00200000` | `Amount` and `LPTokenOut`  | Deposit up to the specified amount of one asset, so that you receive exactly the specified amount of LP Tokens in return (after fees). |
+| `tfLimitLPToken`    | `0x00400000` | `Amount` and `EPrice`      | Deposit up to the specified amount of one asset, but pay no more than the specified effective price per LP Token (after fees). |
+
+Any other combination of these fields and flags is invalid.
+
+
+### Single Asset Deposit Fee Calculations
+
+If you deposit one asset, the AMM charges a fee by reducing the total amount of LP Tokens it pays out, weighted by how much your deposit shifts the balance of assets in the pool. This is equivalent to if you had traded part of the deposit amount for the other asset, then performed a double-asset deposit. The formula for how many LP Tokens you receive for a double-asset deposit is as follows:
```

Comment:
If we make the explanation for single asset deposits more concise in the previous section, we won't have to reiterate what it is here and can instead go straight into the fee calculation.

Reply:
Tweaked this section based on your feedback (see comment above for what I went with)